MekHQ/src/mekhq/campaign/parts/equipment/EquipmentPart.java

```diff
@@ -633,46 +633,24 @@ public boolean isOmniPoddable() {
 
     @Override
     public String getLocationName() {
-        if (null != unit) {
-            Mounted mounted = unit.getEntity().getEquipment(equipmentNum);
-            if (null != mounted && mounted.getLocation() != -1) {
-                return unit.getEntity().getLocationName(mounted.getLocation());
-            }
+        final Mounted mounted = getMounted();
+        if ((mounted != null) && (mounted.getLocation() != Entity.LOC_NONE)) {
+            return getUnit().getEntity().getLocationName(mounted.getLocation());
         }
+
         return null;
     }
 
     @Override
     public boolean isInLocation(String loc) {
```

Comment:
I tried my absolute damnedest to touch the least external code possible.

My last attempts at comprehensive tests seemed to have missed corner cases here and there. This time I have a much higher confidence level because I left the smelly code smelly.